add docs for configuring ghe (#804)
RoadieHQ/marketing-site: content/docs/details/github-enterprise/index.md

---
title: Configuring GitHub Enterprise
publishedDate: '2022-07-28T21:00:00.0Z'
description: How to connect the Roadie catalog to GitHub Enterprise.
---

## Introduction

This tutorial will guide you through the steps required to connect Roadie to your GitHub org in GitHub Enterprise.

## Prerequisites

- Please let the Roadie support team know about your intention to configure GitHub enterprise, because we need to do some manual steps to enable it for you.
- You will need to ensure that the GitHub Enterprise instance is available to our IP addresses published [here](/docs/details/allowlisting-roadie-traffic/): 

## Step 1: Create an OAuth App in GitHub

> Create an OAuth App in GitHub https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app and generate a secret key. Keep note of the client ID and secret key.

Reviewer: Do we need to tell them what scopes to request?

Reviewer: Same for the PAT?

Author: The Oauth app does not have scopes. The PAT token does though.

Reviewer: ah, right! it was the normal GH app 👍  thanks

Author: done.

> Create an OAuth App in GitHub https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app and generate a secret key. Keep note of the client ID and secret key.

Reviewer 2: ```suggestion
Create an [OAuth App](https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app) on GitHub. and generate a secret key. Keep note of the client ID and secret key.
```

## Step 2: Create a personal access token

> Create a personal access token in GitHub with the `repo` permissions. https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token.

Reviewer 2: ```suggestion
Create a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) in GitHub with the `repo` permissions. 
```

## Step 3: Configure the GitHub Enterprise secrets

> Go to the secrets configuration page in Backstage `https://<your-tenant>.roadie.so/administration/settings/secrets` and set secrets `GHE_TOKEN`, `GHE_CLIENT_ID` and `GHE_CLIENT_SECRET`.

Reviewer 2: ```suggestion
Go to the secrets configuration page in Backstage:

`https://<your-tenant>.roadie.so/administration/settings/secrets` 

And add the secrets for `GHE_TOKEN`, `GHE_CLIENT_ID` and `GHE_CLIENT_SECRET`.
```
